sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -310,3 +312,28 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
     defineCodeGen(ctx, ev, c => s"($c).length()")
   }
 }
+
+/**
+ *  A function that return soundex name of a ID
+ */
+case class SoundEx(child: Expression) extends UnaryExpression with ExpectsInputTypes{
+  val so = new Soundex
+  override def dataType: DataType = StringType
+  override def expectedChildTypes: Seq[DataType] = Seq(StringType)
+  override def eval(input: InternalRow): Any = {
+    val string  = child.eval(input)
+    if (string == null) null
+    else {
+      val str: java.lang.String = string.toString
+      so.encode(str)
+
+    }
+  }
+
+  override def toString: String = s"SoundEx($child)"
+
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    defineCodeGen(ctx, ev, c => s"($c).SoundEx()")
```

Comment:
This is not right, and let's remove the `genCode` for now, until we figure out if the codegen will improve the performance.